src/lib.rs

```diff
@@ -1046,3 +889,140 @@ where
+        Ok(Lines {
+            program_rows: program.rows(),
+            sequences: Vec::new(),
+            read_sequences: false,
+        })
+    }
+
+    fn read_sequences(&mut self) {
+        if self.read_sequences {
+            return;
+        }
+        let mut sequences = Vec::new();
+        let mut sequence_rows: Vec<Row> = Vec::new();
+        let mut prev_address = 0;
+        while let Ok(Some((_, &program_row))) = self.program_rows.next_row() {
+            let address = program_row.address();
+            if program_row.address() < prev_address {
+                // The standard says:
+                // "Within a sequence, addresses and operation pointers may only increase."
+                // So this row is invalid, we can ignore it.
+                //
+                // If we wanted to handle this, we could start a new sequence
+                // here, but let's wait until that is needed.
+            } else if program_row.end_sequence() {
+                if !sequence_rows.is_empty() {
+                    let low_address = sequence_rows[0].address;
+                    let high_address = program_row.address();
+                    let mut rows = Vec::new();
+                    std::mem::swap(&mut rows, &mut sequence_rows);
+                    sequences.push(Sequence {
+                        low_address,
+                        high_address,
+                        rows,
+                    });
+                }
+                prev_address = 0;
+            } else {
+                let file_index = program_row.file_index();
+                let line = program_row.line();
+                let mut duplicate = false;
+                if let Some(last_row) = sequence_rows.last_mut() {
+                    if last_row.address == address {
+                        last_row.file_index = file_index;
+                        last_row.line = line;
+                        duplicate = true;
+                    }
+                }
+                if !duplicate {
+                    sequence_rows.push(Row {
+                        address,
+                        file_index,
+                        line,
+                    });
+                }
+                prev_address = address;
+            }
+        }
+        if !sequence_rows.is_empty() {
+            // A sequence without an end_sequence row.
+            // Let's assume the last row covered 1 byte.
+            let low_address = sequence_rows[0].address;
+            let high_address = sequence_rows.last().unwrap().address + 1;
+            sequences.push(Sequence {
+                low_address,
+                high_address,
+                rows: sequence_rows,
+            });
+        }
+        // Sort so we can binary search.
+        sequences.sort_by(|a, b| a.low_address.cmp(&b.low_address));
+        self.sequences = sequences;
+        self.read_sequences = true;
+    }
+
+    fn locate(&self, address: u64) -> Option<&Row> {
```

Comment:
Yep good point. I originally had the `read_sequences()` call here, but it didn't work with the borrow checker.